stockflux-securities-master/src/components/SecuritiesTable.js

```diff
@@ -1,68 +1,68 @@
-  { exch: "NYSE", symbol: "AAPL", name: "Apple inc." },
-  { exch: "STFX", symbol: "SCOTT", name: "Scott Logic inc." },
-  { exch: "STFX", symbol: "SCOTT", name: "Scott Logic inc." },
-  { exch: "LSE", symbol: "AAPL", name: "Apple inc." },
-  { exch: "LSE", symbol: "TSLA", name: "Tesla inc." },
-  { exch: "NYSE", symbol: "AAPL", name: "Apple inc." },
-  { exch: "STFX", symbol: "SCOTT", name: "Scott Logic inc." }
-];
+async function getSecuritiesData() {
+  const response = await fetch("http://ws00128:3001/securities");
+  const securities = await response.json();
+  return securities;
+}
 
 const SecuritiesTable = () => {
+  const [securitiesData, setSecuritiesData] = useState([]);
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getSecuritiesData().then(securities => {
+      setIsLoading(false);
+      setSecuritiesData(securities);
+    }).catch(() => {
+      setIsLoading(false);
+    });
+  }, []);
+
   return (
     <div className="securities-container">
-      <div className="securities-title">My Securities Table</div>
-      <div className="header-container">
-        <div className="securities-table-header">Exchange</div>
-        <div className="securities-table-header">Symbol</div>
-        <div className="securities-table-header">Name</div>
-      </div>
-      <div className="table-body">
-        <Components.ScrollWrapperY>
-          {SECURITIES_DATA.map((item, index) => (
-            <div key={index} className="securities-table-row">
-              <div className="securities-exchange-data">{item.exch}</div>
-              <div className="securities-symbol-data">{item.symbol}</div>
-              <div className="securities-name-data">{item.name}</div>
-            </div>
-          ))}
-        </Components.ScrollWrapperY>
-      </div>
+      <>
+        <div className="securities-title">My Securities Table</div>
+        <div className="header-container">
+          <div className="securities-table-header">Exchange</div>
+          <div className="securities-table-header">Symbol</div>
+          <div className="securities-table-header">Name</div>
+        </div>
+        {isLoading ? (
+          <div className="spinner-container"><Components.LargeSpinner /></div>
+        ) : (
+          <div className="table-body">
+            {securitiesData.length > 0 ? (
+              <Components.ScrollWrapperY>
+                {securitiesData.map((item, index) => (
+                  <div key={index} className="securities-table-row">
+                    <div className="securities-exchange-data">
+                      {item.securityId}
+                    </div>
+                    <div className="securities-symbol-data">{item.symbol}</div>
+                    <div className="securities-name-data">{item.name}</div>
+                  </div>
+                ))}
+              </Components.ScrollWrapperY>
+            ) : (
+              <>
```

Comment:
shouldn't need the fragment wrapper as the container is the parent element

Reply:
okay gotten rid of this now